### PR TITLE
Lidar Decision Node

### LINKED ISSUE #46: Lidar Decision

### PR DESCRIPTION
**Pull request by garethellis0**
- Currently, the node is pretty simple, especially in regards to how the twist message is created (right now it just turns when near an obstacle)
- closes #46

### CODE REVIEW

File: src/decision/src/LidarDecision.cpp [hunk 1]
```diff
@@ -15,22 +16,115 @@ LidarDecision::LidarDecision(int argc, char **argv, std::string node_name) {
...
+    }
+}
+
+geometry_msgs::Twist LidarDecision::twist_message_from_obstacle(LidarObstacle obstacle,
+                                                               distance_t danger_distance,
+                                                               angle_t danger_angle,
+                                                               float linear_vel_multiplier,
+                                                               float angular_vel_multiplier) {
+    geometry_msgs::Twist twist;
+    twist.linear.x = 0;
```
Comment (ValRat) [inline src/decision/src/LidarDecision.cpp]: Would this never make the robot go forward?

File: src/decision/test/lidar_decision_rostest.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    // We would also not want to be going forward
+    EXPECT_EQ(0, command.linear.x);
+
+    // Everything else should always be 0
+    EXPECT_EQ(0, command.linear.y);
+    EXPECT_EQ(0, command.linear.z);
+    EXPECT_EQ(0, command.angular.x);
+    EXPECT_EQ(0, command.angular.y);
+}
+
```
Comment (ValRat) [inline src/decision/test/lidar_decision_rostest.cpp]: Maybe add a test case where there's no obstacles? See how it handles zero input.

File: src/decision/src/LidarDecision.cpp (hunk 1 above)
Comment (garethellis0) [inline src/decision/src/LidarDecision.cpp]: I'm presuming that an all 0 message will be treated by the central decision node as no input (ie. it will either use other decision nodes or just go forward)

**Comment by garethellis0** [comment]
@ValRat another review plz
